src/main/java/Duke.java

```diff
@@ -1,10 +1,603 @@
+import java.util.ArrayList;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+import java.nio.file.Path;
+import java.nio.file.Files;
+
+import exceptions.MarkOutOfBounds;
+import exceptions.UnmarkOutOfBounds;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.FileReader;
+import java.io.BufferedReader;
+import java.io.IOException;
+
 public class Duke {
```

Comment:
Duke class is very long, try extracting to have smaller classes to handle different things.